src/item.cpp

```diff
@@ -3986,11 +3998,27 @@ bool item::ammo_consume( int qty, const tripoint& pos ) {
         return res;
     }
 
-    if( qty > ammo_remaining() ) {
-        return false;
-    }
-
     if( is_magazine() ) {
+        if( type->magazine->linkage != "NULL" ) {
+            item linkage( type->magazine->linkage, bday );
+            linkage.charges = 1; // needs charge 1 to stack properly with other linkages
+
+            auto tiles = closest_tripoints_first( 1, pos );
+            tiles.erase( std::remove_if( tiles.begin(), tiles.end(), [&]( const tripoint& e ) {
+                return !g->m.passable( e ); // avoid walls when ejecting linkages
+            } ), tiles.end() );
+
+            std::random_device rdev;
+            std::default_random_engine engine( rdev() );
+            std::uniform_int_distribution<int> series( 0, tiles.size() - 1 );
```

Comment:
Is our own `rng` function not good enough for you? Actually, there is even `random_entry`, which takes only the container and returns a random entry from it: `eject = random_entry(tiles);`